scene/3d/collision_shape_3d.cpp

```diff
@@ -185,6 +188,7 @@ void CollisionShape3D::set_shape(const Ref<Shape3D> &p_shape) {
 		if (shape.is_valid()) {
 			parent->shape_owner_add_shape(owner_id, shape);
 		}
+		_update_in_shape_owner();
```

Comment:
It looks good, just a question regarding `_update_in_shape_owner();` - why this is needed here? According to `shape_owner_add_shape` it should already updated what is needed.

Reply:
I was trying to make it consistent with 2D and with `NOTIFICATION_PARENTED`, but you're right, it doesn't actually update anything in this case. I will remove it from here and `NOTIFICATION_PARENTED` as well.

Reply:
Done!